asreview/webapp/api/projects.py

```diff
@@ -1529,13 +1612,31 @@ def api_delete_project(project):  # noqa: F401
 
     if project.project_path.exists() and project.project_path.is_dir():
         try:
+
+            # remove from database if applicable
+            if current_app.config["AUTHENTICATION_ENABLED"]:
+                project = Project.query.filter(
+                    and_(
+                        Project.project_id == project.project_id,
+                        Project.owner_id == current_user.id,
+                    )
+                ).one_or_none()
+
+                if project != None:
+                    DB.session.delete(project)
+                    DB.session.commit()
+                else:
+                    return jsonify(message="Failed to delete project in DB."), 500
+
+            # and remove the folder
             shutil.rmtree(project.project_path)
+
         except Exception as err:
```

Comment:
If this exception is triggered, there was an error during `shutil.rmtree`. In this case the project is deleted from the database, but the project folder was not yet deleted. Maybe move the database commit to after this exception handler and add a rollback to the exception handler.